packages/front-end/components/Metrics/MetricForm/index.tsx

```diff
@@ -529,6 +532,21 @@ const MetricForm: FC<MetricFormProps> = ({
     return set;
   }, [value.userIdTypes, type]);
 
+  const validateResponse = (result: TestQueryRow) => {
+    if (!result) return;
+
+    const requiredColumnsArray = Array.from(requiredColumns);
+    const missingColumns = requiredColumnsArray.filter(
+      (col) => !((col as string) in result)
+    );
+
+    if (missingColumns.length > 0) {
+      throw new Error(
+        `You are missing the following columns: ${missingColumns.join(", ")}`
+      );
+    }
+  };
```

Comment:
Move this logic into EditSQL modal and always run when requireColumns are passed in